### ISSUE TITLE
Cannot import goog.modules from an ES6 module

### ISSUE THREAD

**Issue by shicks** [google/closure-compiler#1877]
The following code works if `foo` is a `goog.provide` or a `goog.module` with `goog.module.declareLegacyNamespace`, but fails to compile if it is an ordinary `goog.module`:
```js
import {foo} from 'goog:foo';
```

The error that shows up is
```
es6.js:1: ERROR - required "foo" namespace never provided
import {foo} from 'goog:foo';
```

**Comment by shicks** [comment]
The same thing happens when using
```js
const foo = goog.require('foo');
```
from within an ES6 module.

**Comment by mashedcode** [comment]
Yeah for some reason you need a newline between the destructuring assignment and the `goog.require` so this should work:
```
const { foo } =
  goog.require('foo');
```
I got no clue why science I haven't even looked at the source but that's how I fixed it for my project.

Most of the code generated for ES6 features is quite terrible too IMHO therefore I still prefer `concat` over the spread operator etc. but that's another issue (The destructuring code is fine though). My love for the closure-compiler and it's awesome optimization remains <3.

**Comment by MatrixFrog** [comment]
@mashedcode the newline making any difference at all is very surprising. Are you sure you're on the latest compiler version?

**Comment by mashedcode** [comment]
@MatrixFrog no I wasn't but now I definitely am and it's the same deal.

**Comment by mashedcode** [comment]
@MatrixFrog ohh well (yeah I am now ^^) but I just noticed that it had nothing to do with the closure compiler in my case (sorry for cluttering up this issue).
Because in my case the closurebuilder.py from the closure-library just doesn't get it right when calculating dependencies because of [this super specific regex in source.py](https://github.com/google/closure-library/blob/8bbc453a8808aa77af91f959fb0dc97be058cd97/closure/bin/build/source.py#L31
)
But actually it does compile fine for me @shicks. Maybe you got the same issue?

**Comment by dev-mraj** [comment]
+1

**Comment by MatrixFrog** [comment]
The reasons might be historical so I would suggest trying to send a pull request and see what they say.

**Comment by mashedcode** [comment]
@MatrixFrog @redforks did open a PR (see google/closure-library#739).
It's currently closed because code review comments haven't been addressed for around 3 months.

**Comment by rictic** [comment]
I just ran some basic tests and this seems to be working.

```js
import {foo} from 'goog:foo';
```

Works both when `foo` is a goog.module and when it is goog.provided.
